File: src/contests/cg/puzzle/go/horse-racing-duals.go
// 🎮 CodinGame Puzzle - horse-racing-duals
// https://www.codingame.com/training/easy/horse-racing-duals

package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"sort"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n int
	fmt.Fscan(reader, &n)
	s := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscan(reader, &s[i])
	}
	sort.Ints(s)
	minDiff := math.MaxInt64
	for i := 1; i < n; i++ {
		diff := s[i] - s[i-1]
		if diff < minDiff {
			minDiff = diff
		}
	}
	fmt.Println(minDiff)
}

File: src/contests/cg/puzzle/lua/horse-racing-duals.lua
-- 🎮 CodinGame Puzzle - horse-racing-duals
-- https://www.codingame.com/training/easy/horse-racing-duals

local n = tonumber(io.read())
local s = {}
for i = 1, n do
  s[i] = tonumber(io.read())
end
table.sort(s)
local min_diff = math.huge
for i = 2, n do
  local diff = s[i] - s[i - 1]
  if diff < min_diff then
    min_diff = diff
  end
end
print(min_diff)
